WalletWasabi/Microservices/ProcessBridge.cs

```diff
@@ -26,50 +26,58 @@ public ProcessBridge(string processPath)
-				WindowStyle = windowStyle
-			};
-
+			var process = CreateProcessInstance(arguments, openConsole);
 			try
 			{
-				return Process.Start(startInfo);
+				process.Start();
 			}
 			catch
 			{
-				Logger.LogInfo($"{nameof(startInfo.FileName)}: {startInfo.FileName}");
-				Logger.LogInfo($"{nameof(startInfo.Arguments)}: {startInfo.Arguments}");
-				Logger.LogInfo($"{nameof(startInfo.RedirectStandardOutput)}: {startInfo.RedirectStandardOutput}");
-				Logger.LogInfo($"{nameof(startInfo.UseShellExecute)}: {startInfo.UseShellExecute}");
-				Logger.LogInfo($"{nameof(startInfo.CreateNoWindow)}: {startInfo.CreateNoWindow}");
-				Logger.LogInfo($"{nameof(startInfo.WindowStyle)}: {startInfo.WindowStyle}");
-
+				Logger.LogInfo($"{nameof(process.StartInfo.FileName)}: {process.StartInfo.FileName}");
+				Logger.LogInfo($"{nameof(process.StartInfo.Arguments)}: {process.StartInfo.Arguments}");
+				Logger.LogInfo($"{nameof(process.StartInfo.RedirectStandardOutput)}: {process.StartInfo.RedirectStandardOutput}");
+				Logger.LogInfo($"{nameof(process.StartInfo.UseShellExecute)}: {process.StartInfo.UseShellExecute}");
+				Logger.LogInfo($"{nameof(process.StartInfo.CreateNoWindow)}: {process.StartInfo.CreateNoWindow}");
+				Logger.LogInfo($"{nameof(process.StartInfo.WindowStyle)}: {process.StartInfo.WindowStyle}");
 				throw;
 			}
+			return process;
 		}
 
 		public async Task<(string response, int exitCode)> SendCommandAsync(string arguments, bool openConsole, CancellationToken cancel)
 		{
-			using var process = Start(arguments, openConsole);
-			await process.WaitForExitAsync(cancel).ConfigureAwait(false);
-			string responseString = openConsole ? string.Empty : await process.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+			// "using" makes sure the process exits at the end of this method.
+			using var process = new ProcessAsync(CreateProcessInstance(arguments, openConsole));
+
+			process.Start();
 
+			await process.WaitForExitAsync(cancel);
+
+			string responseString = openConsole ? string.Empty : await process.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
 			return (responseString, exitCode: process.ExitCode);
 		}
+
+		protected Process CreateProcessInstance(string arguments, bool openConsole)
+		{
+			var redirectStandardOutput = !openConsole;
+			var useShellExecute = openConsole;
+			var createNoWindow = !openConsole;
+			var windowStyle = openConsole ? ProcessWindowStyle.Normal : ProcessWindowStyle.Hidden;
+
+			if (openConsole && !RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			{
+				throw new PlatformNotSupportedException($"{RuntimeInformation.OSDescription} is not supported.");
+			}
+
+			var p = new Process();
+			p.StartInfo.FileName = ProcessPath;
+			p.StartInfo.Arguments = arguments;
+			p.StartInfo.RedirectStandardOutput = redirectStandardOutput;
+			p.StartInfo.UseShellExecute = useShellExecute;
+			p.StartInfo.CreateNoWindow = createNoWindow;
+			p.StartInfo.WindowStyle = windowStyle;
```

Comment:
How about this, I removed the redundant `openConsole` check and I inlined the temporary variables.

```suggestion
			ProcessWindowStyle windowStyle;
			if (openConsole)
			{
				if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
				{
					throw new PlatformNotSupportedException($"{RuntimeInformation.OSDescription} is not supported.");
				}

				windowStyle = ProcessWindowStyle.Normal;
			}
			else
			{
				windowStyle = ProcessWindowStyle.Hidden;
			}

			var p = new Process();
			p.StartInfo.FileName = ProcessPath;
			p.StartInfo.Arguments = arguments;
			p.StartInfo.RedirectStandardOutput = !openConsole;
			p.StartInfo.UseShellExecute = openConsole;
			p.StartInfo.CreateNoWindow = !openConsole;
			p.StartInfo.WindowStyle = windowStyle;
```

Reply:
Yeah, that's better. 👍